janusgraph-core/src/main/java/org/janusgraph/graphdb/management/ConfigurationManagementGraph.java

```diff
@@ -0,0 +1,320 @@
+                                    String.format("Your updated template configuration may not contain the property \"%s\".",
+                                                  PROPERTY_GRAPH_NAME
+                                    ));
+        log.warn("Any graph configuration created using the Template_Configuration are only guraranteed to have their configuration updated " +
+                 "according to this new Template_Configuration when the graph in question has been closed on every Janus Graph Node, its " +
+                 "corresponding Configuration has been removed, and the graph has been recreated.");
+        updateVertexWithProperties(propertyTemplate, true, config.getMap());
+    }
+
+
+    /**
+     * Remove Configuration according to {@link String graphName}
+     */
+    public void removeConfiguration(final String graphName) {
+        final Vertex v;
+        try {
+            v = graph.traversal().V().has(PROPERTY_GRAPH_NAME, graphName).next();
+        } catch (NoSuchElementException e) {
+            return;
+        }
+        v.remove();
+        graph.tx().commit();
+    }
+
+    /**
+     * Remove Template_Configuration
+     */
+    public void removeTemplateConfiguration() {
+        final Vertex v;
+        try {
+            v = graph.traversal().V().has(propertyTemplate, true).next();
+        } catch (NoSuchElementException e) {
+            return;
+        }
+        v.remove();
+        graph.tx().commit();
+    }
+
+    /**
+     * Get Configuration according to supplied {@link String} graphName mapped to a specific
+     * {@link Graph}; if does not exist, return null.
+     *
+     * @return Map<String, Object>
+     */
+    public Map<String, Object> getConfiguration(final String configName) {
+        final List<Map<String, Object>> l = graph.traversal().V().has(PROPERTY_GRAPH_NAME, configName).valueMap().toList();
+        if (l.size() == 0) return null;
+        else if (l.size() > 1) { // this case shouldn't happen because our index has a unique constraint
+            log.warn("Your configuration management graph is an a bad state. Please " +
+                     "ensure you have just one configuration per graph. The behavior " +
+                     "of the class' APIs are hereonout unpredictable until this is fixed.");
+        }
+        return deserializeVertexProperties(l.get(0));
+    }
+
+    /**
+     * Get a list of all Configurations, excluding the template configuration; if none exist,
+     * return null
+     *
+     * @return List<Map<String, Object>>
+     */
+    public List<Map<String, Object>> getConfigurations() {
+        final List<Map<String, Object>> l = graph.traversal().V().has(propertyTemplate, false).valueMap().toList();
+        if (l.size() == 0) return null;
+        return l.stream().map(elem -> {
+          return deserializeVertexProperties(elem);
+        }).collect(Collectors.toList());
```

Comment:
`l.stream().map(this::deserializeVertexProperties).collect(Collectors.toList());`